src/js/media-container.js

```diff
@@ -484,11 +454,31 @@ class MediaContainer extends globalThis.HTMLElement {
     this.addEventListener('mouseleave', this);
     this.addEventListener('keyup', this);
 
-    globalThis.window?.addEventListener('mouseup', () => {
-      this.removeAttribute(Attributes.KEYBOARD_CONTROL);
-    });
+    globalThis.window?.addEventListener('mouseup', this);
+  }
+
+  disconnectedCallback() {
+    // When disconnected from the DOM, remove root node and media event listeners
+    // to prevent memory leaks and unneeded invisble UI updates.
+    if (this.media) {
+      this.mediaUnsetCallback(this.media);
+    }
```

Comment:
this fixed the 1st memory leak caused by the global event listener or the event listener to the root node related to fullscreen change.